src/hooks/useEditVideoSheet.tsx

```diff
@@ -180,19 +180,30 @@ export const useEditVideoSheetTabData = (tab?: EditVideoSheetTab) => {
     activeUser: { channelId },
   } = useActiveUser()
   const { drafts } = useDrafts('video', channelId ?? '')
-
   const { selectedVideoTabCachedAssets } = useEditVideoSheet()
 
   const { video, loading, error } = useVideo(tab?.id ?? '', { skip: tab?.isDraft })
+  const [cachedDirtyVideoTabsData, setcachedDirtyVideoTabs] = useState<Array<EditVideoFormFields & EditVideoSheetTab>>(
```

Comment:
I'm a bit worried about this getting out of sync since each hook invocation will have its own state (we don't back this by any context, so will be a separate state for each component that uses this hook). Maybe the data should be persisted by the `EditVideoSheetProvider`? What do you think?

Reply:
Also I'm wondering whether this data should come from this hook versus the general `useEditVideoSheet`, we already return stuff like `setSelectedVideoTabCachedAssets` from it so maybe form state should also be modified from it?

Reply:
Yeah I was a bit confused about where to put it at first; so added it to the provider and moved to `useEditVideoSheet`